src/hooks/screen/useCreateChat.ts

```diff
@@ -7,18 +7,35 @@ import ChannelListMemberSchema from '../../database/schema/ChannelListMemberSche
 import SignedMessageRepo from '../../service/repo/signedMessageRepo';
 import UserSchema from '../../database/schema/UserSchema';
 import useLocalDatabaseHook from '../../database/hooks/useLocalDatabaseHook';
+import useUserAuthHook from '../core/auth/useUserAuthHook';
 import useChatUtilsHook, {AllowedGoToChatScreen} from '../core/chat/useChatUtilsHook';
+import {AnonUserInfoTs} from '../../../types/core/AnonUserInfoTs.types';
 import {GROUP_INFO} from '../core/constant';
+import {MessageAnonymouslyData} from '../../../types/repo/AnonymousMessageRepo/MessageAnonymouslyData';
+import {getChannelListInfo} from '../../utils/string/StringUtils';
 import {getOrCreateAnonymousChannel} from '../../service/chat';
 
 const useCreateChat = () => {
   const [loadingCreateChat, setLoadingCreateChat] = React.useState(false);
   const {localDb} = useLocalDatabaseHook();
   const {goToChatScreen} = useChatUtilsHook();
+  const {signedProfileId, anonProfileId} = useUserAuthHook();
 
   const handleMemberSchema = (response) => {
     try {
-      response?.members?.forEach(async (member) => {
+      const builtChannelData = {
+        better_channel_member: response?.better_channel_member,
+        members: response?.members
+      };
+
+      const {originalMembers} = getChannelListInfo(
+        builtChannelData,
+        signedProfileId,
+        anonProfileId
+      );
+      const members = originalMembers || response?.members;
+
+      members?.forEach(async (member) => {
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

There is a potential issue with the `forEach` loop where you are using an async function. This could lead to unexpected behavior because `forEach` will not wait for promises to resolve before moving on to the next iteration. Consider using `for...of` or `Promise.all()` instead.
```diff
- members?.forEach(async (member) => {
+ for (const member of members) {
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->